server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSDefineOptionsCheckUtility.java

```diff
@@ -0,0 +1,194 @@
+          put("NOSUSPEND", ErrorSeverity.WARNING);
+          put("INPUT", ErrorSeverity.ERROR);
+          put("PROCESS", ErrorSeverity.ERROR);
+          put("PROCESSTYPE", ErrorSeverity.ERROR);
+          put("NOCHECK", ErrorSeverity.WARNING);
+          put("TIMER", ErrorSeverity.ERROR);
+          put("DAYS", ErrorSeverity.ERROR);
+          put("HOURS", ErrorSeverity.ERROR);
+          put("MINUTES", ErrorSeverity.ERROR);
+          put("SECONDS", ErrorSeverity.ERROR);
+          put("YEAR", ErrorSeverity.ERROR);
+          put("MONTH", ErrorSeverity.ERROR);
+          put("DAYOFMONTH", ErrorSeverity.ERROR);
+          put("DAYOFYEAR", ErrorSeverity.ERROR);
+          put("AFTER", ErrorSeverity.ERROR);
+          put("AT", ErrorSeverity.ERROR);
+          put("0N", ErrorSeverity.ERROR);
+        }
+      };
+
+  public CICSDefineOptionsCheckUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS Define rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getClass() == CICSParser.Cics_define_activityContext.class) {
+      checkActivity((CICSParser.Cics_define_activityContext) ctx);
+    } else if (ctx.getClass() == CICSParser.Cics_define_composite_eventContext.class) {
+      checkCompositeEvent((CICSParser.Cics_define_composite_eventContext) ctx);
+    } else if (ctx.getClass() == CICSParser.Cics_define_counter_dcounterContext.class) {
+      checkCounter((CICSParser.Cics_define_counter_dcounterContext) ctx);
+    } else if (ctx.getClass() == CICSParser.Cics_define_input_eventContext.class) {
+      checkInputEvent((CICSParser.Cics_define_input_eventContext) ctx);
+    } else if (ctx.getClass() == CICSParser.Cics_define_processContext.class) {
+      checkDefineProcess((CICSParser.Cics_define_processContext) ctx);
+    } else if (ctx.getClass() == CICSParser.Cics_define_timerContext.class) {
+      checkDefineTimer((CICSParser.Cics_define_timerContext) ctx);
+    }
+    checkDuplicates(ctx);
+  }
+
+  private void checkActivity(CICSParser.Cics_define_activityContext ctx) {
+    checkHasMandatoryOptions(ctx.ACTIVITY(), ctx, "ACTIVITY");
+    checkHasMandatoryOptions(ctx.TRANSID(), ctx, "TRANSID");
+  }
+
+  private void checkCompositeEvent(CICSParser.Cics_define_composite_eventContext ctx) {
+    checkHasMandatoryOptions(ctx.COMPOSITE(), ctx, "COMPOSITE");
+    checkHasMandatoryOptions(ctx.EVENT(), ctx, "EVENT");
+    if (!ctx.OR().isEmpty()) checkHasIllegalOptions(ctx.AND(), "AND");
+    else checkHasMandatoryOptions(ctx.AND(), ctx, "AND or OR");
+  }
+
+  private void checkCounter(CICSParser.Cics_define_counter_dcounterContext ctx) {
+    if (ctx.COUNTER().isEmpty()) checkHasMandatoryOptions(ctx.DCOUNTER(), ctx, "DCOUNTER");
```

Comment:
I think this validation should be done so that the user is told to use "either counter or dcounter" or that you cannot use "both counter and dcounter".